features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/metric/slowframes/UISlownessMetricDispatcher.kt

```diff
@@ -0,0 +1,151 @@
+            return
+        }
+
+        internalLogger.logMetric(
+            samplingRate = samplingRate,
+            messageBuilder = { UI_SLOWNESS_MESSAGE },
+            additionalProperties = buildMetricAttributesMap(
+                slowFramesCount = telemetry.slowFramesCount.get(),
+                ignoredFramesCount = telemetry.ignoredFramesCount.get(),
+                freezeFramesCount = telemetry.freezeFramesCount.get()
+            )
+        )
+    }
+
+    private fun buildMetricAttributesMap(
+        slowFramesCount: Int,
+        ignoredFramesCount: Int,
+        freezeFramesCount: Int
+    ): Map<String, Any> = buildMap {
+        put(KEY_METRIC_TYPE, VALUE_METRIC_TYPE)
+        put(
+            KEY_RUM_UI_SLOWNESS,
+            buildMap {
+                put(
+                    KEY_SLOW_FRAMES,
+                    buildMap {
+                        put(KEY_COUNT, slowFramesCount)
+                        put(KEY_IGNORED_COUNT, ignoredFramesCount)
+                        put(
+                            KEY_CONFIG,
+                            buildMap {
+                                put(KEY_MAX_COUNT, config.maxSlowFramesAmount)
+                                put(KEY_SLOW_FRAME_THRESHOLD, 2.0f) // no option this value to be changed for now
+                                put(KEY_MAX_DURATION, config.maxSlowFrameThresholdNs)
+                                put(KEY_VIEW_MIN_DURATION, config.minViewLifetimeThresholdNs)
+                            }
+                        )
+                    }
+                )
+                put(
+                    KEY_FREEZED_FRAMES,
+                    buildMap {
+                        put(KEY_COUNT, freezeFramesCount)
+                        put(
+                            KEY_CONFIG,
+                            buildMap {
+                                put(KEY_MIN_DURATION, config.freezeDurationThresholdNs)
+                            }
+                        )
+                    }
+                )
+            }
+        )
+    }
+
+    companion object {
+        private const val DEFAULT_SAMPLE_RATE: Float = 0.75f
+
+        internal const val UI_SLOWNESS_MESSAGE = "[Mobile Metric] RUM UI Slowness"
+
+        internal const val KEY_METRIC_TYPE = "metric_type"
+        internal const val VALUE_METRIC_TYPE = "rum ui slowness"
+
+        internal const val KEY_RUM_UI_SLOWNESS = "rum_ui_slowness"
+
+        internal const val KEY_SLOW_FRAMES = "slow_frames"
+        internal const val KEY_COUNT = "count"
+        internal const val KEY_IGNORED_COUNT = "ignored_count"
+        internal const val KEY_CONFIG = "config"
+        internal const val KEY_MAX_COUNT = "max_count"
+        internal const val KEY_SLOW_FRAME_THRESHOLD = "slow_frame_threshold"
+        internal const val KEY_MAX_DURATION = "max_duration"
+        internal const val KEY_VIEW_MIN_DURATION = "view_min_duration"
+
+        internal const val KEY_FREEZED_FRAMES = "freezed_frames"
```

Comment:
Are we sure about the use of "freezed frames" for the plural? I think frozen frames would be more correct

Reply:
We decided to remove frozen frames from the telemetry for now. We gonna add them separately